### PR TITLE
Store plugin OnActivate errors and include them in PluginStatus response

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so we keep track of errors that occur when activating a plugin. We then deliver this error to the client on-demand within the `PluginStatus` struct.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-44999

#### Related Pull Requests

- https://github.com/mattermost/mattermost-webapp/pull/10572

#### Screenshots

image

#### Release Note

```release-note
Plugin activation errors now show in the plugin management page and marketplace
```

### CODE REVIEW

File: plugin/environment.go [hunk 1]
```diff
@@ -213,6 +231,14 @@ func (env *Environment) GetManifest(pluginId string) (*model.Manifest, error) {
 }
 
 func (env *Environment) Activate(id string) (manifest *model.Manifest, activated bool, reterr error) {
+	defer func() {
+		if reterr != nil {
+			env.setPluginError(id, reterr.Error())
+		} else {
+			env.setPluginError(id, "")
```
Comment (cpoile) [inline plugin/environment.go:238]: Are we doing this to clear an error string that might be there from before? (just checking)

**Review by cpoile — APPROVED** [review]

File: plugin/environment.go (hunk 1 above)
Comment (mickmister) [reply plugin/environment.go:238]: @cpoile Yes that's correct. `reterr == nil` means "Successful activation" so we should clear any existing error

**Review by DHaussermann — APPROVED** [review]
Tested and passed
Tested again to ensure this change is working as expected and the error preventing start up is shown. 

This was tested and passed a couple weeks ago. Sorry @mickmister I accidentally approved the webapp PR only and left this open. 

LGTM!

**Pull request merged**